internal/kconfig/kconfig.go

```diff
@@ -263,12 +263,25 @@ func PutInteger(data []byte, integer *btf.Int, n uint64) error {
 		return fmt.Errorf("invalid boolean value: %d", n)
 	}
 
+	if len(data) < int(integer.Size) {
+		return fmt.Errorf("can't fit an integer of size %d into a byte slice of length %d", integer.Size, len(data))
+	}
+
 	switch integer.Size {
 	case 1:
+		if integer.Encoding == btf.Signed && (n > math.MaxInt8 && n < math.MaxUint64-math.MaxInt8) {
+			return fmt.Errorf("can't represent %d as a signed integer of size %d", n, integer.Size)
+		}
 		data[0] = byte(n)
 	case 2:
+		if integer.Encoding == btf.Signed && (n > math.MaxInt16 && n < math.MaxUint64-math.MaxInt16) {
+			return fmt.Errorf("can't represent %d as a signed integer of size %d", n, integer.Size)
+		}
 		internal.NativeEndian.PutUint16(data, uint16(n))
 	case 4:
+		if integer.Encoding == btf.Signed && (n > math.MaxInt32 && n < math.MaxUint64-math.MaxInt32) {
+			return fmt.Errorf("can't represent %d as a signed integer of size %d", n, integer.Size)
+		}
 		internal.NativeEndian.PutUint32(data, uint32(n))
 	case 8:
 		internal.NativeEndian.PutUint64(data, uint64(n))
```

Comment:
This also needs a check, since it's possible for n to overflow/underflow an int64.

Reply:
I'll add some test code for the boundary value of an int32.

Reply:
Never mind this comment, wasn't thinking straight.